temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowService.java

```diff
@@ -572,20 +581,46 @@ public void respondWorkflowTaskFailed(
     }
   }
 
+  private Context.CancellableContext deadlineCtx(final Duration dur) {
+    return Context.current()
+        .withDeadline(Deadline.after(dur.toNanos(), TimeUnit.NANOSECONDS), this.bgSched);
+  }
+
+  /**
+   * Execute blocking function but use the supplied Context to trigger an interrupt of the current
+   * thread in the case that the Context is cancelled.
+   *
+   * @param ctx the context that may be canceled while this thread is blocked in work
+   * @param work the blocking work to perform
+   * @param <T> the result of calling the provided Supplier
+   * @return the result of calling the provided Supplier.
+   */
+  private <T> T withContextInterruption(final Context ctx, final Supplier<T> work) {
```

Comment:
It looks like it should be "withContextCancellation"? because we handle context cancellation here. 
Or "withThreadInterruption"? But this looks wrong. It looks like thread interruption is an implementation detail.